atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTestSchema.java

```diff
@@ -59,6 +59,12 @@ private static Schema generateSchema() {
                     .isAppendHeavyAndReadLight()
                     .build());
 
+        // streams can have up to Integer.MAX_VALUE as the inMemoryThreshold
+        schema.addStreamStoreDefinition(
+                new StreamStoreDefinitionBuilder("stream_test_max_mem", "stream_test_max_mem", ValueType.VAR_LONG)
+                    .inMemoryThreshold(Integer.MAX_VALUE)
```

Comment:
I realize this is a test, but realistically for Hotspot the in memory threshold is just below `Integer.MAX_VALUE`

See http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l239
```
    /**
     * The maximum size of array to allocate (unless necessary).
     * Some VMs reserve some header words in an array.
     * Attempts to allocate larger arrays may result in
     * OutOfMemoryError: Requested array size exceeds VM limit
     */
    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
```